torch_xla/experimental/xla_sharding.py

```diff
@@ -148,6 +148,27 @@ class ShardingSpec:
   mesh: Mesh
   partition_spec: Tuple[Union[int, None]]
 
+  # Derived fields
+  _tile_assignment: List[int] = field(init=False)
+  _replicated: bool = field(init=False)
+  _partial: bool = field(init=False)
+
+  def __post_init__(self):
```

Comment:
Just for my own education, what's the benefit of using __post_init__ v.s. __init__?

Reply:
`dataclass` generates the `__init__` method, which will call into `__post_init__` after initialization.

https://docs.python.org/3/library/dataclasses.html#post-init-processing

Reply:
Then I guess ShardingSpec is no longer a data class then. Anyway, I'm fine with it as it is.